awx/main/dispatch/worker/task.py

```diff
@@ -52,6 +54,8 @@ def run_callable(self, body):
         uuid = body.get('uuid', '<unknown>')
         args = body.get('args', [])
         kwargs = body.get('kwargs', {})
+        if 'guid' in body:
+            GuidMiddleware.set_guid(body.pop('guid'))
```

Comment:
When a (child) dispatcher process begins working on a task, set the guid to the value from the message that came across the bus.